builder/datasource/query.go

```diff
@@ -8,7 +8,7 @@ import (
 
 type Query struct {
 	Base
-	Query builder.Query `json:"-,omitempty"`
+	Query builder.Query `json:"query,omitempty"`
```

Comment:
this make the marshaling works but kill the unmarshaling :face_with_head_bandage: 

Uncomment https://github.com/grafadruid/go-druid/blob/08f180208bf7b61682d08b16ff80c1165a2e59a9/examples/fromjson/main.go#L58 and give it a try

Reply:
Okay, I realized what I needed.
I'll fix it, just be patient.

Reply:
Hi, @jbguerraz  
do you know the history of the following code?
builder/query/query.go
```go
  if b.Type() != "sql" {
      d, err := datasource.Load(tmp.DataSource)
      if err != nil {
	      return err
      }
      if d.Type() == "query" {
	      d.(*datasource.Query).UnmarshalJSONWithQueryLoader(tmp.DataSource, Load)
      }
```
This should originally be implemented as an UnmarshalJSON method in builder/datasource/query.go so it can work properly without the if statement.
But I think it was implemented this way because of the circle import issue.

I fixed the code and all the test code in examples/fromjson/main.go works fine.
```go
    d, err := datasource.Load(tmp.DataSource)
    if err != nil && d.Type() == "query" {
	    err = d.(*datasource.Query).UnmarshalJSONWithQueryLoader(tmp.DataSource, Load) // cycle import
    }
    if err != nil {
	    return err
    }

```

But to fix the underlying problem, you know,
I'm going to have to change a lot of paths in the source code in project.

Reply:
Yes this query loader injection is there to workaround the cycle issue. At the time of implementing the library we didn't find a more elegant way to manage a query that load a query that load a query that load a query that... If you have any suggestion of a better implementation I would love to read it :-)